src/linux/sensor_ds18b20.c

```diff
@@ -217,10 +224,15 @@ ds18_send_and_request(struct ds18_s *d, uint32_t next_begin_time, uint8_t oid)
     if (d->status == W1_ERROR) {
         // try_shutdown expects a static string. Output the specific error,
         // then shut down with a generic error.
-        output("Error: %s", d->error);
         pthread_mutex_unlock(&d->lock);
```

Comment:
The mutex_unlock needs to go inside the conditional. It should stay locked until after the pthread_cond_signal call below, if not shutting down here.